### PR TITLE
CI build and prove tests

### PR DESCRIPTION
**Pull request by jklmnn**
- Travis CI
 - Project with most warnings and errors enabled
 - build with ada-runtime
 - SPARK proof of generic code
 - disabled SPARK on incompatible platform code
 - posix log client
 - dummy posix block implementation
   - dispatcher + server: empty Ada bodies
   - client: empty C binding

### CODE REVIEW

File: src/cai-block-dispatcher.ads [hunk 1]
```diff
@@ -1,7 +1,10 @@
 with Cai.Block.Server;
 
+pragma Warnings (Off, "package ""Serv"" is not referenced");
```
Comment (senier) [inline src/cai-block-dispatcher.ads]: Justify why this warning is suppressed.

File: switches.gpr [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     ,"-gnatyd"    -- check no DOS line terminators
+     ,"-gnatye"    -- check end/exit labels present
+     ,"-gnatyf"    -- check no form feeds/vertical tabs in source
+     ,"-gnatyh"    -- check no horizontal tabs in source
+     ,"-gnatyi"    -- check if-then layout
+     ,"-gnatyI"    -- check mode in
+     ,"-gnatyk"    -- check casing rules for keywords
+     ,"-gnatyl"    -- check reference manual layout
+     ,"-gnatyL8"   -- check max nest level < nn
+     ,"-gnatyM79"  -- check line length <= nn characters
```
Comment (senier) [inline switches.gpr]: M120!

**Comment by jklmnn** [comment]
I adapted the style according to the discussed guidelines.

File: platform/genode/cai-block-client.adb [hunk 3]
```diff
@@ -103,33 +118,36 @@ package body Cai.Block.Client is
...
    begin
       Cxx.Block.Client.Read (
          C.Instance,
-         Util.Convert_Request (R),
+         Client_Util.Convert_Request (R),
          Data);
       B := Convert_Buffer (Data);
    end Read;
 
+   pragma Warnings (Off, "formal parameter ""R"" is not modified");
```
[the remark this answers is not in the document]
Comment (jklmnn) [reply platform/genode/cai-block-client.adb]: The justification has already been added in https://github.com/Componolit/ada-interface/pull/30/commits/1b529791d99cf70e8af93254dfe65fcaf3e4ec83 at L141.

**Pull request merged**